develop/addons/components/genericsetup.rst

```diff
@@ -717,203 +742,1393 @@ Example:
+
+This is the configuration from ``plone.app.contenttypes``:
+
+.. code-block:: xml
+
+    <?xml version="1.0"?>
+    <object>
+      <difftypes>
+        <type portal_type="Document">
+          <field name="any" difftype="Compound Diff for Dexterity types"/>
+        </type>
+        <type portal_type="Event">
+          <field name="any" difftype="Compound Diff for Dexterity types"/>
+        </type>
+        <type portal_type="File">
+          <field name="any" difftype="Compound Diff for Dexterity types"/>
+        </type>
+        <type portal_type="Folder">
+          <field name="any" difftype="Compound Diff for Dexterity types"/>
+        </type>
+        <type portal_type="Image">
+          <field name="any" difftype="Compound Diff for Dexterity types"/>
+        </type>
+        <type portal_type="Link">
+          <field name="any" difftype="Compound Diff for Dexterity types"/>
+        </type>
+        <type portal_type="News Item">
+          <field name="any" difftype="Compound Diff for Dexterity types"/>
+        </type>
+      </difftypes>
+    </object>
+
+This configures how the difference between two versions of a field are shown on the history tab.
+The configuration is stored in the ``portal_diff`` tool.
+
+For Archetypes content, you need a different ``difftype``:
+
+.. code-block:: xml
+
+    <type portal_type="Document">
+      <field name="any" difftype="Compound Diff for AT types"/>
+    </type>
+
+A new ``difftype`` can be registered by calling ``Products.CMFDiffTool.CMFDiffTool.registerDiffType``.
+The ``difftypes`` in standard Plone 5 are:
+
+- ``Lines Diff``
+- ``Compound Diff for AT types``
+- ``Binary Diff``
+- ``Field Diff``
+- ``List Diff``
+- ``HTML Diff``
+- ``Compound Diff for Dexterity types``
+
+.. note::
+
+    There is no uninstall version.
+    The ``remove`` keyword is not supported.
+    The ``portal_diff`` tool does not show configuration for portal_types that no longer exist.
+
+Code is in ``Products.CMFDiffTool.exportimport.difftool``.
+
+
+factorytool.xml
+---------------
+
+This is only needed for Archetypes content types.
+It makes sure when you start adding a content item but don't finish it,
+that no half created item lingers.
+This is not needed for Dexterity items.
+
+.. code-block:: xml
+
+    <?xml version="1.0"?>
+    <object name="portal_factory" meta_type="Plone Factory Tool">
+      <factorytypes>
+        <type portal_type="Document"/>
+        <type portal_type="Event"/>
+        <type portal_type="File"/>
+        <type portal_type="Folder"/>
+        <type portal_type="Image"/>
+        <type portal_type="Link"/>
+        <type portal_type="News Item"/>
+        <type portal_type="Topic"/>
+      </factorytypes>
+    </object>
+
+.. note::
+
+    The ``remove`` keyword is not supported.
+
+Code is in ``Products.ATContentTypes.exportimport.factorytool``.
+
+
+jsregistry.xml
+--------------
+
+.. note::
+
+    This is deprecated in Plone 5.
+
+See :ref:`resourceregistries`.
+
+
```

Comment:
If it is deprecated, does it have to stay, if it a Plone 4 thing, move it there and maybe add a link, if we need to do that?